apollo-relay/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+    }
+}
+
+impl From<tonic::transport::Error> for ReporterError {
+    fn from(error: tonic::transport::Error) -> Self {
+        ReporterError {
+            msg: error.to_string(),
+            source: Box::new(error),
+        }
+    }
+}
+
+impl From<std::io::Error> for ReporterError {
+    fn from(error: std::io::Error) -> Self {
+        ReporterError {
+            msg: error.to_string(),
+            source: Box::new(error),
+        }
+    }
+}
+
+impl From<sys_info::Error> for ReporterError {
+    fn from(error: sys_info::Error) -> Self {
+        ReporterError {
+            msg: error.to_string(),
+            source: Box::new(error),
+        }
+    }
+}
+
+impl From<JoinError> for ReporterError {
+    fn from(error: JoinError) -> Self {
+        ReporterError {
+            msg: error.to_string(),
+            source: Box::new(error),
+        }
+    }
+}
+
+impl std::fmt::Display for ReporterError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "ReporterError: source: {}, message: {}",
+            self.source, self.msg
+        )
+    }
+}
+
+impl Eq for ReporterGraph {}
+
+// PartialEq is derived in the generated code, but Hash isn't and we need
+// it to use this as key in a HashMap. We have to make sure this
+// implementation always matches the derived PartialEq in the generated
+// code.
+#[allow(clippy::derive_hash_xor_eq)]
+impl Hash for ReporterGraph {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.reference.hash(state);
+        self.key.hash(state);
+    }
+}
+
+impl Report {
+    /// Try to create a new Report.
+    ///
+    /// This can fail if the ReportHeader is not created.
+    pub fn try_new(graph: &str) -> Result<Self, ReporterError> {
+        let header = Some(ReportHeader::try_new(graph)?);
+
+        Ok(Report {
+            header,
+            ..Default::default()
+        })
+    }
+}
+
+#[cfg(target_os = "windows")]
+fn get_uname() -> Result<String, std::io::Error> {
+    // Best we can do on windows right now
+    let sysname = sys_info::os_type().unwrap_or("Windows".to_string());
+    let nodename = sys_info::hostname().unwrap_or("unknown".to_string());
+    let release = sys_info::os_release().unwrap_or("unknown".to_string());
+    let version = "unknown";
+    let machine = "unknown";
+    Ok(format!(
+        "{}, {}, {}, {}, {}",
+        sysname, nodename, release, version, machine
+    ))
+}
+
+#[cfg(not(target_os = "windows"))]
+fn get_uname() -> Result<String, std::io::Error> {
+    let u = uname::uname()?;
+    Ok(format!(
+        "{}, {}, {}, {}, {},",
+        u.sysname, u.nodename, u.release, u.version, u.machine
+    ))
+}
+
+impl ReportHeader {
+    fn try_new(graph: &str) -> Result<Self, ReporterError> {
+        let mut header = ReportHeader {
+            ..Default::default()
+        };
+
+        header.add_hostname()?;
+        header.agent_version = std::env!("CARGO_PKG_VERSION").to_string();
```

Comment:
For better or for worse, we use the format `apollo-server-core@1.2.3` for this field, so this should probably contain something specifying the particular library
and not just the version number.

Reply:
Ok. I'm fine with changing this name. I'll make it align with the component name so currently:
```
apollo-relay@0.1.0-alpha.3
```
and will change when component name changes.

Reply:
You now get:
agent_version: "apollo-spaceport-0.1.0-alpha.3"